### PR TITLE
WIP: Add `maxDepth` argument

### PR DESCRIPTION
**Pull request by fleverest**
Todo:
- [x] Implement sampling from the tree with maxDepth
  - [x] EWB case
  - [x] Dirichlet case with leaves having a=a0
  - [x] Write tests for sampling
- [x] ~~Implement sampling leaf probabilities~~
  - [x] ~~Write tests for sampling marginal probabilities, checking that sample mean is close to expected value.~~
- [x] Improve the documentation for the maxDepth functionality

### CODE REVIEW

File: src/RcppIRV.cpp [hunk 1]
```diff
@@ -214,10 +228,13 @@ void PIRVDirichletTree::update(Rcpp::List ballots) {
     // the minDepth of the tree, otherwise the posterior tree
     // will no longer be reducible to a Dirichlet distribution.
     depth = bc.first.nPreferences();
-    if (depth < minDepth && tree->getParameters()->getVD())
-      Rcpp::warning("Updating a Dirichlet distribution with a ballot "
-                    "specifying fewer than `minDepth` preferences. The "
-                    "resulting posterior is no longer Dirichlet.");
+    if (depth < minDepth)
```
Comment (fleverest) [inline src/RcppIRV.cpp]: @dvukcevic The change from `if (depth < minDepth && tree->getParameters()->getVD())` to `if (depth < minDepth)` is the cause of the new warnings. I propose we change it to `if (0 < depth < minDepth)`.

Comment (dvukcevic) [reply src/RcppIRV.cpp]: Sounds like a good idea, @fleverest

**Comment by fleverest** [comment]
I have opted to remove the marginal probability sampling mechanics - the feature was incomplete and I'm honestly not certain it it worked fully, since there was no shared code between the multinomial sampling techniques and the probability sampling used for the marginal probability sampling.

I intend to replace this in the future by adding probability sampling for a set of ballots so that you can draw the correlated probabilities as draws from a single realization of the tree.

File: src/RcppIRV.cpp (hunk 1 above)
Comment (fleverest) [reply src/RcppIRV.cpp]: I've implemented this now.

**Comment by fleverest** [comment]
This PR now also fixes #16 because I rewrote some tests and found it to be useful.

**Comment by fleverest** [comment]
Improved count.ballots (per issue #15) on this version of the package for completeness.

**Pull request merged**